src/solc_0.7/catalyst/AssetMinter.sol

```diff
@@ -137,15 +137,16 @@ contract AssetMinter is WithMetaTransaction {
     function _batchBurnCatalysts(address from, uint256[] memory catalystsQuantities) internal {
         uint16[] memory ids = new uint16[](catalystsQuantities.length);
         for (uint16 i = 0; i < ids.length; i++) {
-            ids[i] = i;
+            ids[i] = i + 1;
```

Comment:
- With `ids[i] = i`, on the first iteration (when i == 0), we're setting the catalyst Id at index `i` in the `ids` array to equal `0` which is not a valid Id for a catalyst.
- We then pass this invalid Id through to `burnCatalyst` eventually, where the execution fails at:
`require(catalyst != Catalyst(0), "CATALYST_DOES_NOT_EXIST");`

Reply:
I now think the way to handle this is instead to make the call to `burnCatalyst` conditional, ie: `if (catalysIds[i] != 0) {...`
rather than changing the logic in `_batchBurnCatalysts`

Reply:
not a bug.